src/Symfony/Component/JsonPath/JsonCrawlerInterface.php

```diff
@@ -22,10 +19,7 @@
 interface JsonCrawlerInterface
```

Comment:
Do we still need this interface?

Reply:
Yes, this is our factory and the interface used to be used in service injection.

One of the next step is adding something like `withCustomFunctions()` that gets a tagged iterator of invokable classes to use in JsonPath filters (the RFC allows custom functions in filters).

Reply:
I wonder though if that doesn't mean that we should rather split the crawler and it's factory. Currently both concerned are mixed in one class which also means that the crawler needs to allow being constructed without data while that shouldn't be needed if the factory was a different class, right?

Reply:
Indeed. We discussed a bit about it with @nicolas-grekas. We agreed that we're fine having everything in one class as well as having a no-op crawler when created without data (i.e. returning an empty array for all paths). Is there a limitation that you see with this implementation maybe?

Reply:
Having the 2 interfaces implemented by the same class is not a problem when using interfaces for typing.
But I'm not convinced by the name of `JsonCrawlerInterface`.

I would prefer `CrawlerFromJsonInterface`.

Reply:
I prefer JsonCrawlerInterface - it feels less convoluted to me, and is explicit about the intended use case - querying json documents.

I also prefer one class to implement the two interfaces. Optimizing the constructor doesn't look worth the split to me.

Reply:
My main concern with mixing responsibilities in one class is that you could have a crawler in an intermediate state while building and then accidentally pass around the wrong crawler instance which would then behave differently than what you did expect. By separating the crawler and its factory (or probably rather a builder) PHP's type system would save you from such a mistake.